src/ast/Location.ts

```diff
@@ -27,13 +28,27 @@ export default class Location extends Node {
     this.address = token;
     
   }  evaluate(context: object[]): void {
-    throw new Error("Method not implemented.");
+    // throw new Error("Method not implemented.");
   }
-  nameCheck(): void {
-    throw new Error("Method not implemented.");
+  
+  nameCheck(map: object): void {
+    if (!map.hasOwnProperty(this.id)) {
+      let values = [null, this.address]; 
+      Object.assign(map, {[this.id as string]: values});
```

Comment:
Similarly to `nameCheck()` for `Guest`, I think we could separate the value storing part into `evaluate()`